mongoengine/document.py

```diff
@@ -429,11 +433,18 @@ def _save_create(self, doc, force_insert, write_concern):
         Helper method, should only be used inside save().
         """
         collection = self._get_collection()
-
-        if force_insert:
-            return collection.insert(doc, **write_concern)
-
-        object_id = collection.save(doc, **write_concern)
+        with set_write_concern(collection, write_concern) as wc_collection:
+            if force_insert:
+                return wc_collection.insert_one(doc).inserted_id
+            # insert_one will provoke UniqueError alongside save does not
+            # therefore, it need to catch and call replace_one.
+            if '_id' in doc:
+                raw_object = wc_collection.find_one_and_replace(
```

Comment:
`find_one_and_replace` has an [upsert](http://api.mongodb.com/python/current/api/pymongo/collection.html#pymongo.collection.Collection.find_one_and_replace) argument, wouldn't that be useful in this case and avoid the necessity to use insert_one)? If yes it would also make it use a single query instead of two

Reply:
insert_one is used for Document which does not have the `_id` field.  Furthermore, the `find_one_and_replace` does not have `upsert` argument. http://api.mongodb.com/python/current/api/pymongo/collection.html#pymongo.collection.Collection.find_one_and_replace

Reply:
Sorry to insist but `upsert` param do exist on that method, it is actually listed in the parameters (I guess its part of **kwargs)